script/print.lua

```diff
@@ -17,14 +17,17 @@ function printmsg(msg, force, useFilter)
     for k, v in pairs(msg) do
       if type(v) == "table" then
         msgKey = msgKey..v[1]..", "
+        msg[k][1] = string.gsub(v[1],"item=","img=item/")
+        msg[k][1] = string.gsub(msg[k][1],"fluid=","img=fluid/")

```

Comment:
v = msg[k] 
doing a lookup with table[key] inside a for loop is just wasting processing time.

further optimization:
gsub can do multiple replacements at once: https://stackoverflow.com/questions/25280374/lua-string-gsub-with-multiple-patterns

Reply:
I knew it wasn't as efficient as it could be and I was just learning LUA. I had implemented changing replacements with gsub as you've pointed out.

I'll take another crack at it.  Thanks for the feedback.

Reply:
Wouldn't the ambiguousness of the icon be a moot point since the player is the one setting the names?

Reply:
Functions should never do blind search and replace of such common stings.
At least printmsg needs another parameter to enable string replacing.
Better yet replacing should happen in the function call.